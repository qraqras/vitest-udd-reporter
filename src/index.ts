export {
	default as VitestUddReporter,
	default,
} from "./vitest-udd-reporter.js";
