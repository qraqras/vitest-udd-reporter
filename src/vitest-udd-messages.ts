const PASSED_MESSAGE: string = `\x1b[32m ユニコォォォーーン！ \x1b[0m`;
const PASSED_MESSAGE_BANNER: string = `\x1b[32m
⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀
⡀⡀⣤⣤⣤⣤⡀⡀⡀⡀⡀⣤⣤⣤⣤⣤⡀⡀⡀⣤⣤⣤⣤⣤⣤⡀⡀⡀⡀⡀⡀⣤⡀⡀⡀⡀⡀⡀⡀⡀⣤⡀⡀⡀⡀⡀⡀⡀⡀⣤⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⣤⣄⡀⡀⡀⡀⡀⡀⡀⡀⡀⣶⡀⡀⡀⡀
⡀⡀⡀⡀⡀⣿⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⣿⡀⡀⡀⣤⣤⣤⣿⣤⣤⡀⡀⡀⣤⣤⣤⣿⣤⣤⡀⡀⡀⣤⣤⣤⣿⣤⣤⡀⡀⣤⣤⣤⣤⣤⣤⣤⣤⡀⣤⣤⣤⣤⣤⣤⣤⣤⡀⡀⡀⡉⡀⡀⡀⣰⡆⡀⡀⡀⡀⣿⡀⡀⡀⡀
⡀⡀⡀⡀⡀⣿⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⣿⡀⡀⡀⡀⣠⡾⣿⡀⡀⡀⡀⡀⡀⣠⡾⣿⡀⡀⡀⡀⡀⡀⣠⡾⣿⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⣀⣾⡋⡀⡀⡀⡀⡀⡛⡀⡀⡀⡀
⡀⡛⡛⡛⡛⡛⡛⡛⡃⡀⡛⡛⡛⡛⡛⡛⡛⡀⡀⡛⡛⡛⡛⡛⡿⡀⡀⡀⡶⡋⣀⡿⡀⡀⡀⡀⡀⡶⡋⣀⡿⡀⡀⡀⡀⡀⡾⡋⣀⡿⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⣶⡶⡿⡋⡀⡀⡀⡀⡀⡀⡀⣶⡀⡀⡀⡀
⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀
\x1b[0m`;
const FAILED_MESSAGE: string = `\x1b[31m それでも！ \x1b[0m`;
const FAILED_MESSAGE_BANNER: string = `\x1b[31m
⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⣠⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⣠⡄⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀
⡀⡀⡘⡛⣩⡿⡉⡀⡀⡀⡀⣸⡀⡀⣀⡀⡀⡀⡀⣤⣤⣤⣴⡶⡶⡛⡀⡀⣀⣀⣸⣀⡀⡀⡀⡀⡀⡀⡀⡰⡶⡀⡀⡀⡀
⡀⣀⣠⣞⣥⣤⣤⡶⡀⡀⡋⣹⣴⡋⡀⡇⡀⡀⡀⡀⡀⣰⡋⡀⣌⡳⡀⡀⡀⡀⣿⡀⡀⡀⡀⡀⡀⡀⡀⡀⡇⡀⡀⡀⡀
⡀⡉⡀⣠⡏⡀⡀⡀⡀⡀⣠⣻⡀⡀⡀⡇⡀⡀⡀⡀⡀⣿⡀⡀⡀⡀⡀⡀⡉⡉⡟⡉⡁⣸⡀⡀⡀⡀⡀⡀⡃⡀⡀⡀⡀
⡀⡀⡀⡸⣦⣀⣀⡀⡀⡘⡁⣸⡀⡀⡀⣧⡾⡁⡀⡀⡀⡈⣷⣤⣄⡀⡀⡀⡀⡀⣷⣀⣀⣸⡀⡀⡀⡀⡀⡰⡶⡀⡀⡀⡀
⡀⡀⡀⡀⡀⡉⡁⡀⡀⡀⡀⡈⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡉⡉⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀⡀
\x1b[0m`;

export const getUddMessages = (useBanner: boolean) => {
	return {
		PASSED_MESSAGE: useBanner ? PASSED_MESSAGE_BANNER : PASSED_MESSAGE,
		FAILED_MESSAGE: useBanner ? FAILED_MESSAGE_BANNER : FAILED_MESSAGE,
	};
};
