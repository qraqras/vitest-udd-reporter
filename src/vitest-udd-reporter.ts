import type { File } from "@vitest/runner";
import type { Reporter, SerializedError, TestModule } from "vitest/node";
import type { TestRunEndReason } from "vitest/reporters";
import { getUddMessages } from "./vitest-udd-messages.js";

interface VitestUddReporterOptions {
	useBanner?: boolean;
}

export default class VitestUddReporter implements Reporter {
	readonly PASSED_MESSAGE: string;
	readonly FAILED_MESSAGE: string;
	message: string;

	constructor(options?: VitestUddReporterOptions) {
		const useBanner = options?.useBanner ?? true;
		const { PASSED_MESSAGE, FAILED_MESSAGE } = getUddMessages(useBanner);
		this.PASSED_MESSAGE = PASSED_MESSAGE;
		this.FAILED_MESSAGE = FAILED_MESSAGE;
		this.message = "";
	}

	async onTestRunEnd(
		_testModules: ReadonlyArray<TestModule>,
		_unhandledErrors: ReadonlyArray<SerializedError>,
		reason: TestRunEndReason,
	) {
		const assertNever = (x: never): never => {
			throw new Error(`Unhandled TestRunEndReason: ${String(x)}`);
		};
		switch (reason) {
			case "passed":
				this.message = this.PASSED_MESSAGE;
				break;
			case "interrupted":
				this.message = this.FAILED_MESSAGE;
				break;
			case "failed":
				this.message = this.FAILED_MESSAGE;
				break;
			default:
				assertNever(reason);
				break;
		}
	}

	async onFinished(_files: File[], _errors: unknown[], _coverage?: unknown) {
		try {
			console.log(this.message);
		} finally {
			this.message = "";
		}
	}
}
