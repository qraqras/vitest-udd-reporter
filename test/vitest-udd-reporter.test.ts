import { afterEach, describe, expect, test, vi } from "vitest";
import { getUddMessages } from "../src/vitest-udd-messages";
import VitestUddReporter from "../src/vitest-udd-reporter";

afterEach(() => {
	vi.restoreAllMocks();
});

const createReporter = (
	opts?: ConstructorParameters<typeof VitestUddReporter>[0],
) => {
	const uddReporter = new VitestUddReporter(opts);
	const logSpy = vi.spyOn(console, "log").mockImplementation(() => {});
	return { uddReporter, logSpy };
};

test("onTestRunEnd sets passed message and onFinished logs it and clears message", async () => {
	const { uddReporter, logSpy } = createReporter();

	await uddReporter.onTestRunEnd([], [], "passed");
	await uddReporter.onFinished([], []);

	expect(logSpy).toHaveBeenCalledWith(uddReporter.PASSED_MESSAGE);
	expect(uddReporter.message).toBe("");
});

test.each(["failed", "interrupted"] as const)(
	'onTestRunEnd("%s") sets failed message and onFinished logs it',
	async (reason) => {
		const { uddReporter, logSpy } = createReporter();

		await uddReporter.onTestRunEnd([], [], reason);
		await uddReporter.onFinished([], []);

		expect(logSpy).toHaveBeenCalledWith(uddReporter.FAILED_MESSAGE);
		expect(uddReporter.message).toBe("");
	},
);

test("constructor respects useBanner option", () => {
	const defaultReporter = new VitestUddReporter();
	const noBannerReporter = new VitestUddReporter({ useBanner: false });

	const defaultMsgs = getUddMessages(true);
	const noBannerMsgs = getUddMessages(false);

	expect(defaultReporter.PASSED_MESSAGE).toBe(defaultMsgs.PASSED_MESSAGE);
	expect(defaultReporter.FAILED_MESSAGE).toBe(defaultMsgs.FAILED_MESSAGE);

	expect(noBannerReporter.PASSED_MESSAGE).toBe(noBannerMsgs.PASSED_MESSAGE);
	expect(noBannerReporter.FAILED_MESSAGE).toBe(noBannerMsgs.FAILED_MESSAGE);
});

// --- Grouped snapshot tests for console output ---
describe("console output snapshots", () => {
	test("passed message with banner", async () => {
		const { uddReporter, logSpy } = createReporter();

		await uddReporter.onTestRunEnd([], [], "passed");
		await uddReporter.onFinished([], []);

		expect(logSpy.mock.calls).toMatchSnapshot();
	});
	test("passed message without banner", async () => {
		const { uddReporter, logSpy } = createReporter({ useBanner: false });

		await uddReporter.onTestRunEnd([], [], "passed");
		await uddReporter.onFinished([], []);

		expect(logSpy.mock.calls).toMatchSnapshot();
	});

	test.each(["failed", "interrupted"] as const)(
		"%s message with banner",
		async (reason) => {
			const { uddReporter, logSpy } = createReporter();

			await uddReporter.onTestRunEnd([], [], reason);
			await uddReporter.onFinished([], []);

			expect(logSpy.mock.calls).toMatchSnapshot();
		},
	);

	test.each(["failed", "interrupted"] as const)(
		"%s message without banner",
		async (reason) => {
			const { uddReporter, logSpy } = createReporter({ useBanner: false });

			await uddReporter.onTestRunEnd([], [], reason);
			await uddReporter.onFinished([], []);

			expect(logSpy.mock.calls).toMatchSnapshot();
		},
	);

	test("no message set (empty string)", async () => {
		const { uddReporter, logSpy } = createReporter();

		// Call onFinished without setting a message first
		await uddReporter.onFinished([], []);

		expect(logSpy.mock.calls).toMatchSnapshot();
	});
});
