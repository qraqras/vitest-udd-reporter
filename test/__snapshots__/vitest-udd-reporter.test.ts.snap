// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`console output snapshots > failed message with banner 1`] = `
[
  [
    "[31m
‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£†‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£†‚°Ñ‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°Ä‚°ò‚°õ‚£©‚°ø‚°â‚°Ä‚°Ä‚°Ä‚°Ä‚£∏‚°Ä‚°Ä‚£Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£§‚£§‚£§‚£¥‚°∂‚°∂‚°õ‚°Ä‚°Ä‚£Ä‚£Ä‚£∏‚£Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°∞‚°∂‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚£Ä‚£†‚£û‚£•‚£§‚£§‚°∂‚°Ä‚°Ä‚°ã‚£π‚£¥‚°ã‚°Ä‚°á‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£∞‚°ã‚°Ä‚£å‚°≥‚°Ä‚°Ä‚°Ä‚°Ä‚£ø‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°á‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°â‚°Ä‚£†‚°è‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£†‚£ª‚°Ä‚°Ä‚°Ä‚°á‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£ø‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°â‚°â‚°ü‚°â‚°Å‚£∏‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°É‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°Ä‚°Ä‚°∏‚£¶‚£Ä‚£Ä‚°Ä‚°Ä‚°ò‚°Å‚£∏‚°Ä‚°Ä‚°Ä‚£ß‚°æ‚°Å‚°Ä‚°Ä‚°Ä‚°à‚£∑‚£§‚£Ñ‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£∑‚£Ä‚£Ä‚£∏‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°∞‚°∂‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°â‚°Å‚°Ä‚°Ä‚°Ä‚°Ä‚°à‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°â‚°â‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä
[0m",
  ],
]
`;

exports[`console output snapshots > failed message without banner 1`] = `
[
  [
    "[31m „Åù„Çå„Åß„ÇÇÔºÅ [0m",
  ],
]
`;

exports[`console output snapshots > interrupted message with banner 1`] = `
[
  [
    "[31m
‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£†‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£†‚°Ñ‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°Ä‚°ò‚°õ‚£©‚°ø‚°â‚°Ä‚°Ä‚°Ä‚°Ä‚£∏‚°Ä‚°Ä‚£Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£§‚£§‚£§‚£¥‚°∂‚°∂‚°õ‚°Ä‚°Ä‚£Ä‚£Ä‚£∏‚£Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°∞‚°∂‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚£Ä‚£†‚£û‚£•‚£§‚£§‚°∂‚°Ä‚°Ä‚°ã‚£π‚£¥‚°ã‚°Ä‚°á‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£∞‚°ã‚°Ä‚£å‚°≥‚°Ä‚°Ä‚°Ä‚°Ä‚£ø‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°á‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°â‚°Ä‚£†‚°è‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£†‚£ª‚°Ä‚°Ä‚°Ä‚°á‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£ø‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°â‚°â‚°ü‚°â‚°Å‚£∏‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°É‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°Ä‚°Ä‚°∏‚£¶‚£Ä‚£Ä‚°Ä‚°Ä‚°ò‚°Å‚£∏‚°Ä‚°Ä‚°Ä‚£ß‚°æ‚°Å‚°Ä‚°Ä‚°Ä‚°à‚£∑‚£§‚£Ñ‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£∑‚£Ä‚£Ä‚£∏‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°∞‚°∂‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°â‚°Å‚°Ä‚°Ä‚°Ä‚°Ä‚°à‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°â‚°â‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä
[0m",
  ],
]
`;

exports[`console output snapshots > interrupted message without banner 1`] = `
[
  [
    "[31m „Åù„Çå„Åß„ÇÇÔºÅ [0m",
  ],
]
`;

exports[`console output snapshots > no message set (empty string) 1`] = `
[
  [
    "",
  ],
]
`;

exports[`console output snapshots > passed message with banner 1`] = `
[
  [
    "[32m
‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°Ä‚£§‚£§‚£§‚£§‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£§‚£§‚£§‚£§‚£§‚°Ä‚°Ä‚°Ä‚£§‚£§‚£§‚£§‚£§‚£§‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£§‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£§‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£§‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£§‚£Ñ‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£∂‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£ø‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£ø‚°Ä‚°Ä‚°Ä‚£§‚£§‚£§‚£ø‚£§‚£§‚°Ä‚°Ä‚°Ä‚£§‚£§‚£§‚£ø‚£§‚£§‚°Ä‚°Ä‚°Ä‚£§‚£§‚£§‚£ø‚£§‚£§‚°Ä‚°Ä‚£§‚£§‚£§‚£§‚£§‚£§‚£§‚£§‚°Ä‚£§‚£§‚£§‚£§‚£§‚£§‚£§‚£§‚°Ä‚°Ä‚°Ä‚°â‚°Ä‚°Ä‚°Ä‚£∞‚°Ü‚°Ä‚°Ä‚°Ä‚°Ä‚£ø‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£ø‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£ø‚°Ä‚°Ä‚°Ä‚°Ä‚£†‚°æ‚£ø‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£†‚°æ‚£ø‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£†‚°æ‚£ø‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£Ä‚£æ‚°ã‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°õ‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°õ‚°õ‚°õ‚°õ‚°õ‚°õ‚°õ‚°É‚°Ä‚°õ‚°õ‚°õ‚°õ‚°õ‚°õ‚°õ‚°Ä‚°Ä‚°õ‚°õ‚°õ‚°õ‚°õ‚°ø‚°Ä‚°Ä‚°Ä‚°∂‚°ã‚£Ä‚°ø‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°∂‚°ã‚£Ä‚°ø‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°æ‚°ã‚£Ä‚°ø‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£∂‚°∂‚°ø‚°ã‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚£∂‚°Ä‚°Ä‚°Ä‚°Ä
‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä‚°Ä
[0m",
  ],
]
`;

exports[`console output snapshots > passed message without banner 1`] = `
[
  [
    "[32m „É¶„Éã„Ç≥„Ç©„Ç©„Ç©„Éº„Éº„É≥ÔºÅ [0m",
  ],
]
`;
